### DISCUSSION TITLE
Advanced Option for Client set ID from Hostname

### DISCUSSION THREAD

**Discussion by xBr0th3rx** [rustdesk/rustdesk-server-pro#483] [Feature Request]
I would like to have the function for our company as an extended function that the client automatically takes the ID of the hostname. 
As we have all clients in our company with a unique ID.

**Comment by CIAIT** [comment]
This!  Identifying devices by their existing Hostname is preferred over the Client ID, if you are deploying this to hundreds of devices (especially if you are deploying from Intune, and the all devices have a patterned name, that your other services use).

**Comment by xBr0th3rx** [comment]
We also have the issue that we have to reinstall the clients. This means that the client then registers again with a new ID. That's why I have to delete old clients in rustdesk server pro every few days because they are created twice. I know the function of registering using the device name from other antivirus programs. There you can set that the client is always registered by the hostname and this is always overwritten because there is only one unique hostname in the company.

That's why such an option would definitely be an advantage, so we wouldn't have any more duplications.

**Comment by rustdesk** [comment]
Will be in rustdesk client 1.4.0 soon.
